### PR TITLE
Testing/outline cypress testing

### LINKED ISSUE #74: Cypress testing outline

### PR DESCRIPTION
**Pull request by KatherineBlaine**
### Files Changed:

All changes were made within the Cypress directory. The following files were added to the e2e folder:
- activities-spec.cy.js
- entry-spec.cy.js
- error-handling-spec.cy.js
- feelings-spec.cy.js
- home-spec.cy.js
- prompt-spec.cy.js

Along with the following fixtures:
- entries.json
- entry.json

### What Has Changed: 

- Each file tests any and all DOM elements! I used the standard cy.get().should().contains() structure, along with .type() and .click() to simulate the user flow.
- I chose to test that the buttons correctly routed the user to the next page and that the URL correctly reflected this navigation at the end of each spec and then visit the specific url route for each spec rather than recreating the entire user flow leading up to that page in the beforeEach of each spec.
- I created fixtures for both GET and POST requests.

### Notes: (includes questions, ideas for the future, anything you want to call out)

- I do not think the POST intercept/fixture is functioning properly. I'm having a hard time wrapping my head around how GET intercepts interact with POST intercepts. The tests are passing, but I don't think it is truly testing our post. If anyone has any ideas let me know!
- After looking into the POST intercept a bit more, the next step is to do error handling testing (test the DOM for the notFound page) and test the PUT request!

### Closes: (any issues that it closes)

closes #74

### CODE REVIEW

File: client/cypress/fixtures/entry.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[{
+  "entry_id": 4,
+  "user_id": 1,
+  "date": "2023-04-14",
+  "feeling": "anxious",
+  "activity": "Go on a sensory walk",    
+  "journal_entry": "Today I am feeling anxious because I have a big project due this weekend"
```
Comment (jfogiato) [inline client/cypress/fixtures/entry.json:7]: 🥲🧡

File: client/cypress/e2e/activities-spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  beforeEach(() => {
+    cy.intercept('POST', 'http://localhost:3001/api/v1/entries', {
+      statusCode: 201,
+      fixture: 'entry.json'
+    })
+    cy.intercept('GET', 'http://localhost:3001/api/v1/entries', {
+      statusCode: 200,
+      fixture: 'entries.json'
+    })
+    cy.visit('http://localhost:3000/what-should-you-do/Anxious')
```
Comment (jfogiato) [inline client/cypress/e2e/activities-spec.cy.js:11]: This is seriously brilliant KB! I would have never thought to do this. Great idea & I'm definitely stealing this for future testing!!

File: client/cypress/e2e/entry-spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  it('Should display the entry completed by the user on this date', () => {
+    cy.get('.journal-text').should('be.visible').contains('Today I am feeling anxious because I have a big test coming up!')
+  })
+
+  it('Should display a button that brings the user from the entry page back to the log page', () => {
+    cy.get('.uni-btn').click()
+    cy.url().should('eq', 'http://localhost:3000/how-you-felt')
+    cy.get('.journal-boundary').should('not.exist')
+    cy.get('.feelingsLog').should('be.visible')
```
Comment (jfogiato) [inline client/cypress/e2e/entry-spec.cy.js:30]: Loving this test - easy to follow, tests everything it needs to test, concise, 🤌

File: client/cypress/e2e/feelings-spec.cy.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      'Valued'
+    ]
+
+    cy.get('.feelings-container').should('be.visible')
+    cy.get('.feelings-container').children().should('have.length', 41)
+
+    feelings.forEach(feeling => {
+      cy.get('.feelings-container').contains(feeling)
+    })
+  })
```
Comment (jfogiato) [inline client/cypress/e2e/feelings-spec.cy.js:71]: BIG BRAIN!!! Would have never thought to use an iterator in my test. Dope!

File: client/cypress/e2e/activities-spec.cy.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  it('Should display a new activity when the user clicks the right arrow', () => {
+    cy.get('.activities-btns').click()
+    cy.get('h2').should('be.visible').contains('Watch, listen to, or read something that will make you laugh.')
+    cy.get('.activity-card > :nth-child(1)').should('be.visible')
+  })
+
+  it('Should display a button that selects the current activity and brings the user from the activities page to the log page', () => {
+    cy.get('.uni-btn').click()
+    cy.get('.activity-page').should('not.exist')
+    cy.url().should('eq', 'http://localhost:3000/how-you-felt')
```
Comment (chrissycooper) [inline client/cypress/e2e/activities-spec.cy.js:34]: Nice work! I love the route testing. Thank you!

**Review by chrissycooper — APPROVED** [review]

File: client/cypress/e2e/home-spec.cy.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    cy.get('.home > :nth-child(2)').should('be.visible').contains('Beam is an interactive self-care feeling tracking app.')
+  })
+
+  it('Should display the home blurb', () => {
+    cy.get('.blurb').should('be.visible').contains('We created a flow intended to help you work through any kind of feeling or emotion you might be experiencing. You\'ll be asked to choose a feeling and reflect on it. Add anything about your experience that might help you avoid or recreate the feeling in the future (depending on the feeling). After your reflection, you\'ll see an offering of a few activities that might help. Your entries will be available to you in the Feelings Archive for future reflection.')
+  })
+
+  it('Should display a final subheading message', () => {
+    cy.get('.home > :nth-child(4)').should('be.visible').contains('We hope this process will help reveal useful patterns over time.')
+  })
```
Comment (jfogiato) [inline client/cypress/e2e/home-spec.cy.js:24]: These tests will definitely fail after my PR for styling is merged. They will be hidden now, and toggled with a button click. I think getting them to pass should be as simple as adding a `cy.get('button').click()` before, but just wanted to call that out!

File: client/cypress/e2e/log-spec.cy.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    cy.get('tbody > :nth-child(2) > :nth-child(2)').should('be.visible').contains('Mad')
+    cy.get('tbody > :nth-child(2) > :nth-child(3)').should('be.visible').contains("Lay on the floor")
+    cy.get('tbody > :nth-child(2) > :nth-child(4)').should('be.visible').contains('📓')
+  })
+
+  it('Should display the feeling and activity the user selected on the specified date within each table column ', () => {
+    cy.get('tbody > :nth-child(3) > :nth-child(1)').should('be.visible').contains('Apr 5, 2023')
+    cy.get('tbody > :nth-child(3) > :nth-child(2)').should('be.visible').contains('Happy')
+    cy.get('tbody > :nth-child(3) > :nth-child(3)').should('be.visible').contains('Call someone you love.')
+    cy.get('tbody > :nth-child(3) > :nth-child(4)').should('be.visible').contains('📓')
```
Comment (jfogiato) [inline client/cypress/e2e/log-spec.cy.js:40]: Appreciate the thorough-city here KB!

File: client/cypress/support/commands.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
```
Comment (jfogiato) [inline client/cypress/support/commands.js:12]: ur mom's a parent command

File: server/knexfile.js [hunk 9]
```diff
@@ -9,8 +9,8 @@ module.exports = {
     client: 'pg',
     connection: {
       port: 5432,
-      user: "joefogiato",
```
Comment (jfogiato) [inline server/knexfile.js:12]: RIP me

**Review by jfogiato — APPROVED** [review]
So so so appreciative of your work on this KB! Super thorough tests, intuitive testing design, readable, and robust! Thank you!!

File: client/cypress/e2e/home-spec.cy.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+  it('Should display the site logo', () => {
+    cy.get('.logo-home').should('be.visible').contains('BEAM')
+  })
+
+  it('Should display a heading for the site info blurb' , () => {
+    cy.get('.home > :nth-child(2)').should('be.visible').contains('Beam is an interactive self-care feeling tracking app.')
+  })
+
+  it('Should display the home blurb', () => {
```
[the remark this answers is not in the document]
Comment (jfogiato) [reply client/cypress/e2e/home-spec.cy.js:18]: @chrissycooper great minds

File: client/cypress/e2e/activities-spec.cy.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+  it('Should display the header and the activity page', () => {
+    cy.get('.activity-page').should('be.visible')
+    cy.get('header').should('be.visible')
+  })
+
+  it('Should display an activity card that shows one activity idea at a time', () => {
+    cy.get('.activity-card').should('be.visible')
+    cy.get('h2').should('be.visible').contains('Take a long shower or bath, and change into clothes that make you feel comfortable and confident!')
+    cy.visit('http://localhost:3000/what-should-you-do/Anxious')
```
Comment (espressoGoddess) [inline client/cypress/e2e/activities-spec.cy.js:22]: I can't quite tell what this line is doing here at the end of the test?

File: client/cypress/e2e/activities-spec.cy.js (hunk 5 above)
Comment (espressoGoddess) [reply client/cypress/e2e/activities-spec.cy.js:34]: agreed!

**Review by espressoGoddess — APPROVED** [review]
Nice thorough work here!! Curious about the css changes here! I am happy to take a look at the Post intercepts later today!

**Pull request merged**